security/backends/elasticsearch/writer.py

```diff
@@ -40,14 +41,24 @@
 MAX_VERSION = 9999
 
 
-def batch_delete_queryset(queryset, batch):
-    tmp_params = queryset._params.copy()
-    queryset._params['max_docs'] = batch
-    queryset._params['refresh'] = True
-    while True:
-        if queryset.delete()['deleted'] == 0:
-            break
-    queryset._params = tmp_params
+def batch_delete_queryset(queryset, batch, stdout, stderr, attempt=0):
+    try:
+        tmp_params = queryset._params.copy()
+        queryset._params['max_docs'] = batch
+        queryset._params['refresh'] = True
+        queryset._params['request_timeout'] = settings.ELASTICSERACH_CLEAN_LOGS_DELETE_REQUEST_TIMEOUT
```

Comment:
now we have to set timeout to 10s on production for every elasticsearch request because of this query. Is better set higher request timeout only for this delete.